fzf-tab.zsh

```diff
@@ -306,7 +308,24 @@ _fzf_tab_complete() {
 
 zle -C _fzf_tab_complete complete-word _fzf_tab_complete
 
+_fzf_tab_try_custom_completion() {
+    # do not steal fzf's completions
+    [[ $LBUFFER =~ ${(q)FZF_COMPLETION_TRIGGER-'**'}$ ]] && return 1
```

Comment:
I set `FZF_COMPLETION_TRIGGER=""` and bind `fzf-completion` to a separate key (Ctrl-T). This way I can use TAB for normal completions and Ctrl-T to complete file arguments with fzf. I find that pressing Ctrl-T to complete file arguments with fzf is more convenient than pressing \*-\*-TAB.

With this setup `_fzf_tab_try_custom_completion` always bails out early.

Does the check on the first line of `_fzf_tab_try_custom_completion` presuppose that `_fzf_tab_orig_widget` is `fzf-completion` (or some other widget that respects `FZF_COMPLETION_TRIGGER`)? I set `_fzf_tab_orig_widget` to `expand-or-complete`, so this check doesn't make much sense.

FWIW, I don't know what's the right thing to do here. I'm unsure how exactly I want completions to work. Just wanted to share that the current code is a bit odd and likely incorrect.

Reply:
I just checked for this configuration:
```
export FZF_COMPLETION_TRIGGER=""
bindkey '^T' fzf-completion
bindkey '^I' fzf-tab-complete

export fzf_default_completion=expand-or-complete
export _fzf_tab_orig_widget=expand-or-complete
```

If no custom completion function is defined, TAB triggers fzf-tab, ctrl-t triggers fzf.
If some custom completion function is defined, both TAB and ctrl-t triggers it.

>_fzf_tab_try_custom_completion always bails out early
You mean it return 1 on the first if?
```
$ FZF_COMPLETION_TRIGGER=""
$ echo ${(q)FZF_COMPLETION_TRIGGER-'**'}
''
```
so if I am correct, everything should match..

Reply:
> You mean it return 1 on the first if?

Yes.

> If some custom completion function is defined, both TAB and ctrl-t triggers it.

Are you sure TAB triggers it? It doesn't for me. When I press TAB, `_fzf_tab_try_custom_completion` returns on the first line, so `_fzf_complete_*` is never called. Can you verify that `_fzf_tab_orig_widget` is indeed `expand-or-complete` in your setup?